### ISSUE TITLE
Allow wrapping `ReduceLROnPlateau` for LR schedulers

### ISSUE THREAD

**Issue by alanhdu** [PyTorchLightning/pytorch-lightning#7553]
## 🚀 Feature

Right now, support for `ReduceLROnPlateau` with lightning is hard-coded using a special `reduce_on_plateau` flag:

https://github.com/PyTorchLightning/pytorch-lightning/blob/53f8d9a8005121472af8d56f1b333e83d95ece35/pytorch_lightning/trainer/optimizers.py#L150-L152

It would be nice to figure out some way of allowing wrappers around `ReduceLROnPlateau` as well.

### Motivation

GradualWarmupScehduler (from https://github.com/ildoonet/pytorch-gradual-warmup-lr) is a LR scheduler which warms up the learning rate for some set amount of epochs before delegating to another LR scheduler, which seems to be useful for training with very large batches. I would like to combine this with `ReduceLROnPlateau`, but there doesn't seem a great way to plumb this through to Lightning.

I could write a special subclass of `ReduceLROnPlateau` that combines the `ReduceLROnPlateau` and `GradualWarmupScheduler`, but that breaks some of the modularity of `GradualWarmupScheduler`. It'd be great if there was a way to signal to Ligthning to pass in the right metrics arguments to the LR scheduler, even if it's not a `ReduceLROnPlateau` scheduler.

**Comment by tchaton** [comment]
Hey @alanhdu,

I think this could be done with some refactoring internally and by inspecting the step function of the scheduler.

Here is the function doing the step: https://github.com/PyTorchLightning/pytorch-lightning/blob/ad9118f04afe7e9453418adc8298dee0b5868a14/pytorch_lightning/trainer/connectors/optimizer_connector.py#L30

```
                if lr_scheduler['reduce_on_plateau']:
                    lr_scheduler['scheduler'].step(**inspected_dict)
                else:
                    lr_scheduler['scheduler'].step()
```

I think the template should potentially also be revamped.

```
                lr_dict = {
                    'scheduler': lr_scheduler, # The LR scheduler instance (required)
                    'interval': 'epoch', # The unit of the scheduler's step size
                    'frequency': 1, # The frequency of the scheduler
                    'step_keys': {"epoch": "internal", "metrics": "monitor"}, # Inject arguments to `GradualWarmupScheduler`
                    'reduce_on_plateau': False, # For ReduceLROnPlateau scheduler
                    'monitor': 'val_loss', # Metric for ReduceLROnPlateau to monitor
                    'strict': True, # Whether to crash the training if `monitor` is not found
                    'name': None, # Custom name for LearningRateMonitor to use
                }
``` 

Best,
T.C

**Comment by alanhdu** [comment]
That makes sense, but unfortunately I don't think I'll be able to prioritize this in the near-future. I can circle back to this in a month or so, but I wouldn't want to block anyone if they're interested in doing this.

**Comment by alanhdu** [comment]
No update on my end. I would like to get to this "at some point", but I'm clawing through a large backlog at work and probably be able to anytime soon.

**Comment by alanhdu** [comment]
Could we reopen this? I think this is still something we want (even if I don't personally have time to take this on).

**Comment by sevmag** [comment]
Has this ever been addressed? Is there a way to use a ReduceLRonPlateau with warmup in lightning?
